docs/components/page-higher-order/withHeadingManager.tsx

```diff
@@ -2,13 +2,13 @@ import * as React from 'react';
 
 import { HeadingManager } from '~/common/headingManager';
 
-export const HeadingsContext = React.createContext<HeadingManager>(null);
+export const HeadingsContext = React.createContext<HeadingManager | null>(null);
 
 const withHeadingManager = <P extends object>(
   Component: React.ComponentType<P & { headingManager: HeadingManager }>
 ) => (props: P) => (
   <HeadingsContext.Consumer>
-    {headingManager => <Component headingManager={headingManager} {...props} />}
+    {headingManager => <Component headingManager={headingManager!} {...props} />}
```

Comment:
Interesting 🤔 
You declared `export const HeadingsContext = React.createContext<HeadingManager | null>(null);`, so the initial value would be `null` and then in render function you call `headingManager!`? Something is off here 🤔 
Either `<Component />` can actually accept `headingManger={null}` or this render function is never called if the `headingManger` is null 🤔

Reply:
Heading manager is never null when that render is called, because all components are wrapped in `<Provider/>` with actual instance, but `createContext()` requires some default value which cannot be provided at this time.